### DISCUSSION TITLE
Lint specific rule + auto-add ignore comments

### DISCUSSION THREAD

**Discussion by Faithfinder** [biomejs/biome#2618] [Ideas]
Started a new job recently and have opened the new to me world of Python. Where I met the local counterpart to Biome - [Ruff](https://docs.astral.sh/ruff/). And it has a couple of very nice capabilities that work together really well:

* `ruff check --select %RULE%`  would lint the codebase using that rule only
* `ruff check --add-noqa` would automatically add ignore comments to every violation

Combined together, it means that you can easily evaluate a new rule for your codebase, and either auto-fix that exact rule (safe or unsafe fixes), or ignore all existing occurrences, all without messing with the config. I loved the experience and would love something like that in JS world

**Comment by Conaclos** [comment]
I proposed the first one in #58. I will work on in a near future.

**Comment by mistercrunch** [comment]
Found this while looking for a`ruff  check --add-noqa` to help justify a biome migration from eslint in apache/superset. It's a killer feature for large codebases, especially in conjunction with `--ignore-noqa`, where you can still work on rules old lints you decided to disregard in that way at a point in time.

It allows for "stopping the bleeding" without having to pay the price of linting the entire codebase. Set the rules you wish you had today and they're applied moving forward. It's a super key feature for a linter!

**Reply by roan-scytale** [reply]
Chiming in as another ruff user who really valued that approach. Especially on larger code bases, going and adding hundreds of `biome-ignore` comments is infeasible without some automation

**Reply by dyc3** [reply]
This exists already. See the `--suppress` option. See also: https://biomejs.dev/linter/#migrate-from-other-linters

**Reply by mistercrunch** [reply]
Oh nice. This pushes me over the edge in attempting a migration. I'll try and report back here about how it went if that's useful.
